src/ServiceControl.Monitoring/Timings/LongValueOccurrenceHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Monitoring.Timings
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using Processing.RawData.NServiceBus.Metrics;
+    using Processing.Snapshot;
+
+    class TimingsReportHandler : IHandleMessages<LongValueOccurrences>
+    {
+        readonly TimingsStore processingTimeStore;
+        readonly CriticalTimeStore criticalTimeStore;
+
+        static string ProcessingTimeMessageType = "NServiceBus.Metrics.ProcessingTime";
+        static string CriticalTimeMessageType = "NServiceBus.Metrics.CriticalTime";
+
+        public TimingsReportHandler(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore)
+        {
+            this.processingTimeStore = processingTimeStore;
+            this.criticalTimeStore = criticalTimeStore;
+        }
+
+        public Task Handle(LongValueOccurrences message, IMessageHandlerContext context)
+        {
+            var messageType = context.MessageHeaders[Headers.EnclosedMessageTypes];
```

Comment:
Should we use a different header for these? Or should it be a part of the payload to send the vector type in? If someone changes the inheritance hierarchy of this message type, the code will break (silently)